static/js/brand-store/Members/MembersTable.js

```diff
@@ -1,8 +1,61 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import PropTypes from "prop-types";
 import { MainTable, CheckboxInput } from "@canonical/react-components";
 
-function MembersTable({ filteredMembers }) {
+function MembersTable({ filteredMembers, changedMembers, setChangedMembers }) {
+  const [members, setMembers] = useState(filteredMembers);
+
+  const handleRoleChange = (member, role) => {
+    const updatedMembers = members.map((item) => {
+      let updatedItem = {};
+
+      if (item.id === member.id) {
+        if (item.roles.includes(role)) {
+          updatedItem = {
+            ...item,
+            roles: item.roles.filter((r) => r !== role),
+          };
+        } else {
+          updatedItem = {
+            ...item,
+            roles: [].concat(item.roles, [role]),
+          };
+        }
+
+        if (changedMembers.find((m) => m.id === member.id)) {
+          if (
+            JSON.stringify([...member.roles].sort()) !==
```

Comment:
While this is a tricky way of comparing two arrays it's not very resilient to values that won't stringify. In fact it'll make the application 🔥 . So I'd recommend using an existing library for this or creating a little util like...

untested...
```js
function checkArrayEqual(array1, array2) {
  if (array1.length !== array2.length) {
    return false
  }

  const array2Sorted = array2.roles.slice().sort();
  return array1.slice().sort().every((val, idx) => val === array2Sorted[idx]);
}

checkArrayEqual(member.roles, updatedItem.roles);
```

Reply:
Done